transformations/replace_abbreviation_and_acronyms/token.py

```diff
@@ -0,0 +1,258 @@
+    | Types are mutually exclusives.
+
+    | Those types are the basic ones that should be detected and could be replaced easily in a basic augmentation
+    """
+    OTHER = 1
+    ABBREVIATION_EXPANDED = 2
+    ABBREVIATION_ACRONYM = 3
+
+
+class Token:
+    """
+    | A token is the smallest irreducible part of a sentence
+    | It can be a word, a punctuation mark, a suite of numeric characters
+    """
+
+    def __init__(self, token: str or Dict, is_nlp: bool):
+        """
+        if is_nlp is True, the token is a dictionary creates from parsing a string using Spacy NLP library
+        """
+        self.type = None
+        if is_nlp:
+            self.string = token.text_with_ws
+            self.tag = token.tag_  # useless?
+        else:
+            self.string = token
+            self.tag = None
+
+    def replace(self, replacement: str):
+        """
+        Replaces the token by another string
+        """
+        self.string = replacement
+        if self.type == Type.ABBREVIATION_EXPANDED:
+            self.type = Type.ABBREVIATION_ACRONYM
+        elif self.type == Type.ABBREVIATION_ACRONYM:
+            self.type = Type.ABBREVIATION_EXPANDED
+
+
+class AbbreviationTransformation:
+    def __init__(self, seed=0, max_outputs=1, abbreviation_file: str = "abbreviations.txt", preprocessing: bool = False):
+        self.max_outputs = max_outputs
+        self.seed = seed
+        self.abbreviations = load_abbreviation_file(abbreviation_file)
+        self.preprocessing = preprocessing
+
+    def can_be_replaced(self, idx: int, token_list: [Token]) -> bool:
+        """
+        Checks if the word can be replaced by an abbreviation.
+        """
+        abbreviations = [element.lower() for [_, element] in self.abbreviations]
+        if (
+            token_list[idx].tag is not None
+            and get_wordnet_pos(token_list[idx].tag) != wordnet.VERB
+        ):
+            return token_list[idx].string.strip().lower() in [
+                element.lower() for [element, _] in self.abbreviations
+            ] or token_list[idx].string.strip().lower() in [
+                element.split()[0] for element in abbreviations
+            ]
+        return False
+
+    def identify_abbreviation_type(self, index: int, token_list: [Token]) -> [Token]:
+        """
+        Retrieves the type of the token if it is an abbreviation (Type.ABBREVIATION) or (Type.OTHER)
+        """
+        _token = token_list[index]
+        if _token.string.strip().lower() in [
+            element.lower() for [element, _] in self.abbreviations
+        ]:
+            _token.type = Type.ABBREVIATION_ACRONYM
+            return token_list
+        abbreviations = [element.lower().split() for [_, element] in self.abbreviations]
+        potential_abbreviation_indexes = [
+            _token.string.strip().lower() == element[0] for element in abbreviations
+        ]
+        for counter, has_same_token_start in enumerate(potential_abbreviation_indexes):
```

Comment:
this whole for loop is kinda heavy and not easy to read. would u mind reworking it?